gradle-consistent-versions-idea-plugin/src/test/java/com/palantir/gradle/versions/intellij/VersionPropsLexerTests.java

```diff
@@ -0,0 +1,167 @@
+                "has.end.of:line=comments # here is the comment",
+                " has.extra.space:bad_token = 1",
+                "has . space . around:dots = 1",
+                "# A comment",
+                "has.spaces.around : colon = 1",
+                "has.no.spaces.around:equals=1",
+                "has.loads.of.spaces:around   =   equals",
+                "    # comment after spaces"
+            })
+    @ParameterizedTest
+    public void test_psi_tree_structure(String input) throws Exception {
+        JavaCodeInsightTestFixture fixture = getFixture();
+        fixture.configureByText("versions.props", input);
+
+        Optional<String> inCi = Optional.ofNullable(System.getenv("CI"));
+        ApplicationManager.getApplication().runReadAction(() -> {
+            PsiFile psiFile = fixture.getFile();
+            String tokenString = convertPsiToString(psiFile);
+
+            if (inCi.equals(Optional.of("true"))) {
+                String expectedTokens = loadExpectedTokens(input);
+                assertThat(tokenString).isEqualTo(expectedTokens);
+            } else {
+                saveActualTokens(input, tokenString);
+            }
+        });
+    }
+
+    private String convertPsiToString(PsiFile psiFile) {
+        StringBuilder tokenStringBuilder = new StringBuilder();
+        psiFile.accept(new PsiRecursiveElementVisitor() {
+            @Override
+            public void visitElement(PsiElement element) {
+                super.visitElement(element);
+                tokenStringBuilder
+                        .append(element.getNode().getElementType())
+                        .append(" ")
+                        .append(element.getNode().getText())
+                        .append("\n");
+            }
+        });
+        return tokenStringBuilder.toString().trim();
+    }
+
+    private String loadExpectedTokens(String fileName) {
+        try (BufferedReader reader = Files.newBufferedReader(getOutputFilePath(fileName), StandardCharsets.UTF_8)) {
+            return reader.lines().collect(Collectors.joining("\n"));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void saveActualTokens(String fileName, String token) {
+        try (BufferedWriter writer = Files.newBufferedWriter(getOutputFilePath(fileName), StandardCharsets.UTF_8)) {
+            writer.write(token);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Path getOutputFilePath(String input) throws IOException {
+        String fileName = hashToFolderName(input);
+        Path outputPath = Paths.get("src/test/resources/lexerTests/" + fileName);
+        Files.createDirectories(outputPath.getParent());
+        return outputPath;
+    }
+
+    private static String hashToFolderName(String input) {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
```

Comment:
You can use Guava here: `Hashing.sha256().hashString(input, StandardCharsets.UTF_8).toString()`